src/groups.jl

```diff
@@ -4,7 +4,15 @@
 
 immutable BenchmarkGroup
     tags::Vector{Any}
-    data::Dict{Any,Any}
+    data::Associative
+
+    function BenchmarkGroup(tags, data::Dict)
+        new(convert(Vector{Any}, tags), convert(Dict{Any, Any}, data))
+    end
+
+    function BenchmarkGroup(tags, data)
+        new(convert(Vector{Any}, tags), data)
+    end
 end
```

Comment:
That seems backwards incompatible -- there are some tests that construct other dicts (e.g. `Dict{String, Trial}` and `Dict{String, TrialEstimate}`) and [some code](https://github.com/JuliaCI/BenchmarkTools.jl/blob/master/src/groups.jl#L49) that assume `.data` is `{Any, Any}`.